src/Raven.Server/Monitoring/ServerLimitsMonitor.cs

```diff
@@ -0,0 +1,214 @@
+                    if (_logger.IsOperationsEnabled)
+                    {
+                        _logger.Operations($"Running close to OS limits detected:{Environment.NewLine}" + warningMsg);
+                    }
+
+                    var alert = AlertRaised.Create(
+                        null,
+                        "Running close to OS limits",
+                        warningMsg,
+                        AlertType.LowServerLimits,
+                        NotificationSeverity.Warning);
+
+                    _notificationCenter.Add(alert);
+                }
+            }
+            catch (Exception e)
+            {
+                if (_logger.IsOperationsEnabled)
+                    _logger.Operations($"Failed to run {nameof(ServerLimitsMonitor)}", e);
+            }
+            finally
+            {
+                _alerts.Clear();
+                Monitor.Exit(_runLock);
+            }
+        }
+
+        private void CheckPidMaxLimits()
+        {
+            var maxPidMax = GetMaxValueForLimit(pid_max);
+
+            using var proc = new Process();
+            proc.StartInfo.RedirectStandardOutput = true;
+            proc.StartInfo.RedirectStandardError = true;
+            proc.StartInfo.FileName = "ps";
+            proc.StartInfo.Arguments = @"-efL";
+            proc.Start();
+
+            proc.WaitForExit();
+
+            if (proc.ExitCode != 0)
+            {
+                throw new InvalidOperationException($"Executing '{proc.StartInfo.FileName} {proc.StartInfo.Arguments}' failed (exit code {proc.ExitCode}){Environment.NewLine}StandardOutput:{Environment.NewLine}{proc.StandardOutput.ReadToEnd()}{Environment.NewLine}StandardError:{Environment.NewLine}{proc.StandardError.ReadToEnd()}");
+            }
+
+            var currentPidMax = GetEolCount(proc.StandardOutput.BaseStream);
+
+            AddAlertIfNeeded(currentPidMax, maxPidMax, MaxPidMaxThreshold, pid_max, nameof(pid_max));
+        }
+
+        private void CheckThreadsMaxLimits()
+        {
+            var maxThreadsMax = GetMaxValueForLimit(threads_max);
+
+            var currentThreadsMax = 0L;
+            foreach (var _ in Directory.EnumerateDirectories("/proc/self/task", "*", SearchOption.TopDirectoryOnly))
+            {
+                currentThreadsMax++;
+            }
+
+            AddAlertIfNeeded(currentThreadsMax, maxThreadsMax, MaxThreadsMaxThreshold, threads_max, nameof(threads_max));
+        }
+
+        private void CheckMaxMapCountLimits()
+        {
+            var maxMapCount = GetMaxValueForLimit(max_map_count);
+
+            long currentMapCount = GetCurrentMapCount();
+            AddAlertIfNeeded(currentMapCount, maxMapCount, MaxMapCountThreshold, max_map_count, nameof(max_map_count));
+        }
+
+        private static long GetCurrentMapCount()
+        {
+            long currentMapCount;
+            using (FileStream stream = File.OpenRead("/proc/self/maps"))
```

Comment:
What if the file doesn't exist? (Should always do on modern systems, but what if we are running on BSD, etc?)
Let's do error handling at the method level, instead and return a default value on error